File: coding-test/java-kotlin/src/kr/co/programmers/java/practice/CreateWeirdChars.java
package kr.co.programmers.java.practice;

public class CreateWeirdChars {

    public static void main(String[] args) {
        String s = "try hello world";
        System.out.println(new Solution().solution(s));
    }

    static class Solution {
        public String solution(String s) {
            String answer = "";

            String[] sArr = s.split("");
            int idx = -1;
            for(String str : sArr){
                idx = str.equals(" ") ? -1 : idx + 1;
                answer += idx % 2 == 0 ? str.toUpperCase() : str.toLowerCase();
            }
            return answer;
        }
    }
}

File: coding-test/javascript/src/practice/create-weird-chars.js
// @ts-check

/**
 *
 * @param {string} s
 * @returns {string}
 */
function solution(s) {
  let answer = ''
  let idx = 1
  for (let i = 0; i < s.length; i += 1) {
    const charat = s.charAt(i)
    if (charat === ' ') {
      answer += charat
      idx = 1
    } else {
      answer += idx % 2 === 0 ? charat.toLowerCase() : charat.toUpperCase()
      idx += 1
    }
  }

  return answer
}

const s = 'try hello world'
console.log(solution(s))
